ci-operator/step-registry/cucushift/installer/check/ibmcloud/kms-key/cucushift-installer-check-ibmcloud-kms-key-commands.sh

```diff
@@ -24,31 +24,94 @@ critical_check_result=0
-
-#check that node os disk is encrypted
-machines=$(oc get machines -A --no-headers | awk '{print $2}')
-if [[ ! $(echo "${machines}" | wc -l) -gt 0 ]]; then
-  echo "ERROR: Fail to find machines ${machines}"
-  exit 1
+if [[ -n ${id_m} ]]; then
+    mapfile -t vols_m < <(ibmcloud kp registrations -i ${id_m} -o JSON  | jq -r .[].resourceCrn)
+    echo "INFO: master key registrations list is: ${#vols_m[@]}" "${vols_m[@]}"
+    #check that node os disk is encrypted
+    machines=$(oc get machines -A --no-headers | grep master | awk '{print $2}')
+    if [[ ! $(echo "${machines}" | wc -l) -gt 0 ]]; then
+        echo "ERROR: Fail to find master machines ${machines}"
+        critical_check_result=1
+    fi
+    echo "check the master nodes..."
+    for machine in ${machines}; do
+        echo "--- check machine ${machine} ---"
+        volCrn=$(ibmcloud is instance ${machine} --output JSON | jq -r .boot_volume_attachment.volume.crn)
+        #shellcheck disable=SC2076    
+        if [[ -z "${volCrn}" ]] || [[ ! " ${vols_m[*]} " =~ " ${volCrn} " ]]; then
+            echo "ERROR: fail to find the volumn ${volCrn} of ${machine} in the registration list!"
+            critical_check_result=1
+        fi
+    done
 fi
 
-for machine in ${machines}; do
-    echo "--- check machine ${machine} ---"
-    volCrn=$(ibmcloud is instance ${machine} --output JSON | jq -r .boot_volume_attachment.volume.crn)
-    #shellcheck disable=SC2076    
-    if [[ -z "${volCrn}" ]] || [[ ! " ${vols[*]} " =~ " ${volCrn} " ]]; then
-        echo "ERROR: fail to find the volumn ${volCrn} of ${machine} in the registration list!"
+if [[ -n ${id_w} ]]; then
+    mapfile -t vols_w < <(ibmcloud kp registrations -i ${id_w} -o JSON  | jq -r .[].resourceCrn)
+    echo "INFO: worker key registrations list is: ${#vols_w[@]}" "${vols_w[@]}"
+    machines=$(oc get machines -A --no-headers | grep worker | awk '{print $2}')
+    if [[ ! $(echo "${machines}" | wc -l) -gt 0 ]]; then
+        echo "ERROR: Fail to find worker machines ${machines}"
         critical_check_result=1
     fi
-done
+    echo "check the worker nodes..."
+    for machine in ${machines}; do
+        echo "--- check machine ${machine} ---"
+        volCrn=$(ibmcloud is instance ${machine} --output JSON | jq -r .boot_volume_attachment.volume.crn)
+        #shellcheck disable=SC2076    
+        if [[ -z "${volCrn}" ]] || [[ ! " ${vols_w[*]} " =~ " ${volCrn} " ]]; then
+            echo "ERROR: fail to find the volumn ${volCrn} of ${machine} in the registration list!"
+            critical_check_result=1
+        fi
+    done
+fi
 
+if [[ -n ${id_d} ]]; then
+    mapfile -t vols_d < <(ibmcloud kp registrations -i ${id_d} -o JSON  | jq -r .[].resourceCrn)
+    echo "INFO: default key registrations list is: ${#vols_d[@]}" "${vols_d[@]}"
+    vols=$(ibmcloud is vols --encryption user_managed | grep data | awk '{print $1}')
+    if [[ ! $(echo "${machines}" | wc -l) -gt 0 ]] && [[ ${#vols_d[@]} -gt 0 ]] ; then
```

Comment:
here I think we do not need to check machines, because the above master and worker machines are already checked.